pennylane/qnode.py

```diff
@@ -350,14 +350,21 @@ def construct(self, args, kwargs=None):
 
         # quantum circuit function return validation
         if isinstance(res, pennylane.operation.Observable):
-            self.output_type = float
+            if res.return_type == "sample":
+                # Squeezing ensures that there is only one array of values returned
+                # when only a single-mode sample is requested
+                self.output_type = np.squeeze
+            else:
+                self.output_type = float
+
             self.output_dim = 1
             res = (res,)
         elif isinstance(res, Sequence) and res and all(isinstance(x, pennylane.operation.Observable) for x in res):
             # for multiple observables values, any valid Python sequence of observables
             # (i.e., lists, tuples, etc) are supported in the QNode return statement.
             self.output_dim = len(res)
             self.output_type = np.asarray
+
```

Comment:
```suggestion
```

Reply:
I think not having newlines at the end of the file gives CodeFactor hickups, no?